src/python_testing/matter_testing_support.py

```diff
@@ -707,6 +709,19 @@ def check_pics(self, pics_key: str) -> bool:
         pics_key = pics_key.strip().upper()
         return pics_key in picsd and picsd[pics_key]
 
+    def openCommissioningWindow(self, dev_ctrl: ChipDeviceCtrl, node_id: int) -> CommissioningParameters:
+        rnd_discriminator = random.randint(0, 4095)
+        try:
+            commissioning_params = dev_ctrl.OpenCommissioningWindow(nodeid=node_id, timeout=900, iteration=1000,
+                                                                    discriminator=rnd_discriminator, option=1)
+            discriminator_dict = {"random_discriminator": rnd_discriminator}
+            params = (commissioning_params, discriminator_dict)
+
+            return params
```

Comment:
return is `CommisioningParameters` and that is a class with 3 members. You cannot return a tuple here or you have to change the return type.

Consider validating your types using `mypy`

Reply:
In my first try, I went more deeper and I notice that `CommissioningParameters` like you said is a class with three (properties) and is not easy modify a tuple just like that so I have created another class e.g.:

```python
@dataclass
class CustomCommissioningParameters:
    commissioningParameters: CommissioningParameters
    randomDiscriminator: int
```

So instead return commissioningParams and dictionary, I just return CustomCommissioningParameters that include both.

Reply:
This is fine, but i think Andrei was just referring to the return type of the function.

Q: why can commissioning parameters not be modified?